### PR TITLE
Lw/gql client

### PR DESCRIPTION
**Pull request by hey-waffles**

### CODE REVIEW

File: src/components/authetication/AuthenticationCard.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React, { useState } from "react";
 import { Card, Button, Form } from "react-bootstrap";
 import { useRouter } from "next/router";
+import { emailLogin } from "../../utilities/auth";
+import LoginForm2 from "./LoginForm";
```
Comment (hey-waffles) [inline src/components/authetication/AuthenticationCard.tsx:5]: Rename this to something more appropriate. LoginForm is fine, LoginSection should be used below

File: src/components/authetication/AuthenticationCard.tsx [hunk 2]
```diff
@@ -51,37 +55,31 @@ function LoginForm(props: any) {
...
-  function emailLogin() {
-    fetch("/api/account/email-login", {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify({
-        username: user,
-        password: pass
-      })
-    }).then(() => {router.push(router.pathname)});
+  function submitLogin() {
```
Comment (hey-waffles) [inline src/components/authetication/AuthenticationCard.tsx:58]: Remove this; unused

File: src/components/authetication/AuthenticationCard.tsx [hunk 3]
```diff
@@ -51,37 +55,31 @@ function LoginForm(props: any) {
   const [user, setUser] = useState("");
```
Comment (hey-waffles) [inline src/components/authetication/AuthenticationCard.tsx:55]: Remove these

File: src/components/authetication/AuthenticationCard.tsx [hunk 4]
```diff
@@ -51,37 +55,31 @@ function LoginForm(props: any) {
...
-        onClick={() => { props.setSession({ "user": { isLoggedIn: true } }); }}
-      >
+      <Button variant="secondary">
         Google Login
       </Button>
 
-      <Form>
+      <LoginForm2/>
+
+      {/* <Form>
```
Comment (hey-waffles) [inline src/components/authetication/AuthenticationCard.tsx:72]: Remove commented code

File: src/components/authetication/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { Formik, Form as FormikForm, ErrorMessage, } from "formik";
+import * as Yup from "yup";
+import { emailLogin } from "../../utilities/auth";
+import { useRouter } from "next/router";
+import { Col, Form, Row, Button } from "react-bootstrap";
+import { Input } from "../design/forms/Forms";
+
+interface LoginValues {
```
Comment (hey-waffles) [inline src/components/authetication/LoginForm.tsx:9]: Comment this page!

File: src/components/authetication/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+const validationSchema = Yup.object({
+  email: Yup.string()
+    .required("Required"),
+  password: Yup.string()
+    .required("Required"),
+});
+
+export default function LoginForm() {
```
Comment (hey-waffles) [inline src/components/authetication/LoginForm.tsx:26]: :O No comment!

File: src/pages/index.tsx [hunk 7]
```diff
@@ -19,21 +21,21 @@ import { getSession } from "../utilities/auth";
...
 
   // TODO - move the session up to App
-  const [session, setSession] = React.useState({
-    "user": {
-      "isLoggedIn": false,
-      "username": "laura",
-      "displayName": "Laura",
-    },
-  });
+  // const [session, setSession] = React.useState({
```
Comment (hey-waffles) [inline src/pages/index.tsx:24]: Delete commented code

File: src/pages/index.tsx [hunk 8]
```diff
@@ -19,21 +21,21 @@ import { getSession } from "../utilities/auth";
...
+  // });
 
   // Sets the view for the currenly logged in user
-  let userView: JSX.Element;
-  if (session.user.isLoggedIn) {
-    userView = <UserView session={session} setSession={setSession} />;
-  } else {
-    userView = <GuestView session={session} setSession={setSession} />;
-  }
+  const userView: JSX.Element = <GuestView/>;
```
Comment (hey-waffles) [inline src/pages/index.tsx:33]: Delete commented code

File: src/utilities/graphql/realmClient.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * onError link that will attempt to get a fresh access token and retry
+ * a failed request when an Unauthorized error is encountered
+ */
+const refreshLink = onError(({ networkError, operation, forward }) => {
+  if (networkError?.message == "Response not successful: Received status code 401") {
+    // Let's refresh token through async request
+    // Apollo observables are needed to do async inside an onError link
+    console.log("Unauthorized, getting refresh token");
```
Comment (hey-waffles) [inline src/utilities/graphql/realmClient.ts:37]: Remove console.log

**Comment by lucyawrey** [comment]
Everything looks good, merging

**Pull request merged**